src/RenderWebGL.js

```diff
@@ -360,24 +384,176 @@ class RenderWebGL extends EventEmitter {
+                // is bigger than the given ordering, or the ordering is null.
+                // This will place the item according to the ordering or before the first un-ordered
+                // (ordering property is null) item.
+                const insertIndex = currentDrawables.findIndex(
+                    item => (item.ordering > inGroupOrdering) || item.ordering === null);
+                const newDrawableOrdering = {
+                    drawableID: drawableID,
+                    ordering: inGroupOrdering
+                };
+                if (insertIndex < 0) {
+                    // None of the orderings in the layerGroup were bigger than the current
+                    // or null, add the new drawableOrdering desc to the end of the list
+                    currentDrawables.push(newDrawableOrdering);
+                } else {
+                    currentDrawables.splice(insertIndex, 0, newDrawableOrdering);
+                }
+            } else {
+                // inGroupOrdering is undefined or invalid
+                // Set a new ordering for the new drawable:
+                // 0, for empty array, or 1 + last ordering in array
+                const newOrdering = currentDrawables.length > 0 ?
+                    currentDrawables[currentDrawables.length - 1].ordering + 1 : 0;
+                // Add the drawable ordering desc with new ordering to the end of the group
+                currentDrawables.push({
+                    drawableID: drawableID,
+                    ordering: newOrdering
+                });
+            }
+        } else {
+            // Implicitly ordered group
+            if (typeof inGroupOrdering === 'number') {
+                log.warn(`Explicit ordering, ${inGroupOrdering}, provided for implicitly ordered group.` +
+                    `Ignoring ordering and placing drawable at the front.`);
+            }
+            currentDrawables.push(drawableID);
+        }
+
+        // Update full draw list
+        this._updateLayers();
+    }
+
+    _updateLayers () {
+        this._drawList = [];
+        for (let i = 0; i < this._groupOrdering.length; i++) {
+            const currGroupName = this._groupOrdering[i];
+            const currentLayerGroup = this._layerGroups[currGroupName];
+            // All the layer groups should already be sorted according to explicit or implicit
+            // ordering, get an array of just the drawableIDs
+            const layerGroupDrawables = currentLayerGroup.explicitOrdering ?
+                currentLayerGroup.drawables.map(item => item.drawableID) :
+                currentLayerGroup.drawables;
+            this._drawList = this._drawList.concat(layerGroupDrawables);
+        }
+        if (this._unspecifiedLayerGroup) { // Add unspecified items at the very end
+            this._drawList.concat(this._unspecifiedLayerGroup.drawables);
+        }
+    }
```

Comment:
Have you tested the performance of this? It looks innocent at first but with the `map` inside the `for` loop we're at `O(n²)`-ish and it could in theory run once per clone in a project that hits the clone limit, so, like, `O(n³)`...
That said, our clone limit isn't huge so maybe it's fine?

If it does turn out to be slow it might be enough to just do the cache + dirty flag thing...